packages/ngtools/webpack/src/compiler_host.ts

```diff
@@ -79,19 +86,24 @@ export class WebpackCompilerHost implements ts.CompilerHost {
 
   invalidate(fileName: string): void {
     const fullPath = this.resolve(fileName);
-
-    if (this._memoryHost.exists(fullPath)) {
-      this._memoryHost.delete(fullPath);
-    }
-
     this._sourceFileCache.delete(fullPath);
 
     try {
       const exists = this._syncHost.isFile(fullPath);
       if (exists) {
         this._changedFiles.add(fullPath);
       }
-    } catch { }
+    } catch {
```

Comment:
`isFile` may or may not throw if the file exists.  Some host do and some do not.
Maybe something similar to the following:
```
let exists = false;
try {
  exists = this._syncHost.isFile(fullPath);
  if (exists) {
    this._changedFiles.add(fullPath);
  }
} catch { }

// File doesn't exist anymore, so we should delete the related virtual files.
if (!exists && fullPath.endsWith('.ts')) {
  this._virtualFileExtensions.forEach(ext => {
    const virtualFile = (fullPath.slice(0, -3) + ext) as Path;
    if (this._memoryHost.exists(virtualFile)) {
      this._memoryHost.delete(virtualFile);
    }
  });
}
```